slingr-stack/pandadoc-package
# #13746 Migration from endpoint to package

Pull request for the topic https://github.com/slingr-stack/platform/issues/13746 created by @pasaperez-slingr 

## What does it do?

This ticket implements the process of migrating helpers and Java code from PandaDoc Endpoint https://github.com/slingr-stack/pandadoc-endpoint to a package.
It creates the package.json descriptor, migrates the standardized Java code to Javascript scripts and functions for use with Rhino, exports helper functions, updates Readme documentation and creates a listener to replace the endpoint default. 
This package has a dependency on the http service.

## How to test it?

* The packete must be registered in the Developer Portal. 
* Then in the builder of an App, which is internal, the http-service must be created (it works in both shared and dedicated cloud). At the moment the dependency is handled manually.
* In the builder of an App, you must create the PandaDoc package, with version v0.0.36 or higher if any. (You need to connect to a PandaDoc App with Oauth or use an API token) Api token test: "b6e6c576b16263f0639303058e542ea419e58a52fc".
* push changes.

* in the console test with the command:
```
log(JSON.stringify(pkg.pandadoc.functions.documents.get()))
```
You should get a response with a list that currently contains a PandaDoc Sample file, with its id and attributes. 
* You can test with the different pandadoc helpers.

* you can create an Action in an Entity with the following code:
```
  var result = pkg.pandadoc.functions.documents.get();
  var name = 'Documents: ' + (result.results[0].name);
  var id = 'Documents: ' + (result.results[0].id);
  sys.logs.warn(JSON.stringify(pkg.pandadoc.functions.documents.get(id)))
  record.field('attriuteName').val(name);
  sys.data.save(record);
```

## Implementation notes

Create new version of the package v1.0.0.

## Deployment notes

N/A
